app/src/main/java/com/example/taskyy/data/repositories/AgendaRepositoryImpl.kt

```diff
@@ -138,6 +150,27 @@ class AgendaRepositoryImpl @Inject constructor(
         TODO("Not yet implemented")
     }
 
+    @RequiresApi(Build.VERSION_CODES.O)
+    override suspend fun startWorkManager(context: Context) {
+        val constraints = Constraints.Builder()
+            .setRequiredNetworkType(NetworkType.CONNECTED)
+            .build()
+        val workRequest = PeriodicWorkRequestBuilder<AgendaItemWorker>(
+            repeatInterval = 2,
+            repeatIntervalTimeUnit = TimeUnit.MINUTES
+        ).setBackoffCriteria(
+            backoffPolicy = BackoffPolicy.LINEAR,
+            duration = Duration.ofSeconds(15)
+        ).setConstraints(constraints)
+            .build()
+        val workManager = WorkManager.getInstance(context)
+        workManager.enqueueUniquePeriodicWork(
+            "AgendaItemWorker",
+            ExistingPeriodicWorkPolicy.KEEP,
+            workRequest
+        )
+    }
```

Comment:
This is already much better as what you had before. Ideally, I'd put this in a separate class in the data layer, since scheduling such workers isn't really the typical job of a repository, but you can also leave this here